src/uct/ib/base/ib_iface.h

```diff
@@ -188,11 +188,11 @@ uct_ib_iface_invoke_am(uct_ib_iface_t *iface, uint8_t am_id, void *data,
     void *desc = (char*)ib_desc + iface->config.rx_headroom_offset;
     ucs_status_t status;
 
-    status = uct_iface_invoke_am(&iface->super, am_id, data, length, desc);
+    uct_recv_desc_iface(desc) = &iface->super.super;
```

Comment:
but why need to move it before calling the callback? for example, in MM code it's still after the callback